### PR TITLE
@alloy: Marketing Signup Modal component

### PR DESCRIPTION
**Pull request by craigspaeth**
This creates a component that's used for collector acquisition. For details on the behavior see https://github.com/artsy/force/issues/365. The long-term goal is to make these more customizable for Marketing to swap out images and copy for all sorts of campaigns.

image

### CODE REVIEW

File: apps/fair/client/index.coffee [hunk 1]
```diff
@@ -1,9 +1,9 @@
 _ = require 'underscore'
 Backbone = require 'backbone'
 sd = require('sharify').data
+MarketingSignupModal = require '../../../components/marketing_signup_modal/index.coffee'
 initFairLayout = require '../../../components/fair_layout/client/index.coffee'
 FairBrowseRouter = require '../components/browse/router.coffee'
-# FairInfoView = require './info.coffee'
```
Comment (alloy) [inline apps/fair/client/index.coffee]: What’s the point of leaving these in?

File: apps/fair/client/index.coffee [hunk 2]
```diff
@@ -87,3 +86,6 @@ module.exports.init = ->
           "&profile_id=#{profile.id}" +
           "&fair_id=#{fair.id}" +
           "&fair_name=#{fair.get('name')}"
+
+  # Add marketing signup CTA modal
+  new MarketingSignupModal
```
Comment (alloy) [inline apps/fair/client/index.coffee]: Wow, what does it mean in CoffeeScript for a line like this to be at the end of a file on its own?

File: components/marketing_signup_modal/index.coffee [hunk 3]
```diff
@@ -0,0 +1,52 @@
+Backbone = require 'backbone'
+url = require 'url'
+sd = require('sharify').data
+FlashMessage = require '../flash/index.coffee'
+modalize = require '../modalize/index.coffee'
+template = -> require('./index.jade') arguments...
```
Comment (alloy) [inline components/marketing_signup_modal/index.coffee]: What. Are. These. Arguments you speak of… o_O ?

File: config.coffee [hunk 4]
```diff
@@ -113,6 +113,9 @@ module.exports =
   EDITORIAL_CTA_BANNER_IMG: 'http://files.artsy.net/images/iphone_email.png'
   ARTSY_PARTNER_UPDATES_CHANNEL: '5762d454b5989e6f98f7799a'
   TEAM_BLOGS: '^\/life-at-artsy$|^\/artsy-education$|^\/gallery-insights$|^\/artsy-partner-updates$'
+  MARKETING_SIGNUP_MODAL_H1: 'Sign up'
+  MARKETING_SIGNUP_MODAL_H2: 'For early access'
+  MARKETING_SIGNUP_MODAL_IMG: 'http://placekitten.com/700/1000'
```
Comment (alloy) [inline config.coffee]: Does this lead to an actual request?

File: apps/fair/client/index.coffee (hunk 1 above)
Comment (craigspaeth) [inline apps/fair/client/index.coffee]: I removed these commented lines, I think it was accidentally left from someone else.

File: components/marketing_signup_modal/index.coffee (hunk 3 above)
Comment (craigspaeth) [inline components/marketing_signup_modal/index.coffee]: :-P `arguments` are a keyword in Javascript. This line is basically wrapping the template in an anonymous function to defer it's parsing to runtime.

This is an awkwardly useful way to unit test these browserified files in Node. Admittedly there's probably a better way to do this, maybe running the code through Browserify before requiring it in tests.

**Comment by craigspaeth** [comment]
Thanks for the review!

Coffeescript looks fairly foreign at first, but it's not all that much different from JS (especially the latest versions). Although I'm excited to be moving on to ES6 going forward, I encourage taking a look through the [Coffescript docs](http://coffeescript.org/). It's a quick read and will be helpful in navigating the Coffeescript in Artsy's Backbone/Node and Rails projects.

**Comment by craigspaeth** [comment]
Going to merge for @briansw to QA

**Comment by alloy** [comment]
Thanks! Will give that a read.

**Pull request merged**